vortex-array/src/array/sparse/mod.rs

```diff
@@ -152,10 +152,22 @@ impl SparseArray {
             let min_index: usize = scalar_at(self.indices(), 0)
                 .and_then(|s| s.as_ref().try_into())
                 .vortex_expect("SparseArray indices is non-empty");
-
             min_index - self.indices_offset()
         })
     }
+
+    /// Return the maximum index if indices are present.
+    ///
+    /// If this sparse array has no indices (i.e. all elements are equal to fill_value)
+    /// then it returns None.
+    pub fn max_index(&self) -> Option<usize> {
```

Comment:
I know I added min_index but we shouldn't pile more here. The methods you want here is first_non_null and last_non_null and that can be implemented for any array